app/components/BoggleCards.tsx

```diff
@@ -0,0 +1,221 @@
+"use client";
+
+import { Box, Button, TextField, Card } from "@mui/material";
+import { indigo, lightGreen, blue, green } from "@mui/material/colors";
+import { keyframes } from "@mui/system";
+
+import { useState } from "react";
+import WordList from "./WordsList";
+
+interface Props {
+  squareGrid: string[][];
+  word: string;
+  setWord: (word: string) => void;
+  words: string[];
+  setWords: (words: string[]) => void;
+  pressedTiles: number[][];
+  setPressedTiles: (pressedTiles: number[][]) => void;
+  errMessage: string;
+  setErrMessage: (message: string) => void;
+}
+
+export default function BoggleCards(props: Props) {
+  const {
+    squareGrid,
+    words,
+    setWords,
+    word,
+    setWord,
+    pressedTiles,
+    setPressedTiles,
+    errMessage,
+    setErrMessage,
+  } = props;
+
+  const [lastCharPos, setLastCharPos] = useState<number[]>([]);
+
+  function createWord(char: string, location: number[]) {
+    if (invalidTileTapped(location))
+      return handleError("Tile already selected!");
+
+    setWord(word + char);
```

Comment:
Do we need both `word` state and `pressedTiles` state? Could we just get the word from the `pressedTiles` state?